src/coreComponents/linearAlgebra/interfaces/hypre/SuperluUtils.cpp

```diff
@@ -0,0 +1,175 @@
+ */
+
+#include "SuperluUtils.hpp"
+#include "common/Stopwatch.hpp"
+
+namespace geosx
+{
+
+void ConvertToSuperMatrix( HypreMatrix const & matrix,
+                           array1d< globalIndex > & rowPtr,
+                           array1d< globalIndex > & cols,
+                           array1d< real64 > & vals,
+                           SuperMatrix & SLUDMat )
+{
+  localIndex const numLocalRows = matrix.numLocalRows();
+  rowPtr.resize( numLocalRows+1 );
+  cols.reserve( matrix.numLocalNonzeros() );
+  vals.reserve( matrix.numLocalNonzeros() );
+  rowPtr[0] = 0;
+  for( localIndex i = 0; i < numLocalRows; ++i )
+  {
+    localIndex const nonZeros = matrix.localRowLength( i );
+    array1d< globalIndex > colIndices( nonZeros );
+    array1d< real64 > values( nonZeros );
+    matrix.getRowCopy( matrix.getGlobalRowID( i ), colIndices, values );
+    cols.insert( rowPtr[i], colIndices.begin(), colIndices.end() );
+    vals.insert( rowPtr[i], values.begin(), values.end() );
+    rowPtr[i+1] = rowPtr[i] + nonZeros;
+  }
+
+  dCreate_CompRowLoc_Matrix_dist( &SLUDMat,
+                                  toSuperlu_intT( matrix.numGlobalRows() ),
+                                  toSuperlu_intT( matrix.numGlobalRows() ),
+                                  toSuperlu_intT( matrix.numLocalNonzeros() ),
+                                  toSuperlu_intT( numLocalRows ),
+                                  toSuperlu_intT( matrix.ilower() ),
+                                  vals.data(),
+                                  toSuperlu_intT( cols.data() ),
+                                  toSuperlu_intT( rowPtr.data() ),
+                                  SLU_NR_loc,
+                                  SLU_D,
+                                  SLU_GE );
+}
+
+int SolveSuperMatrix( SuperMatrix & SLUDMat,
+                      HypreVector const & b,
+                      HypreVector & x,
+                      MPI_Comm const & comm,
+                      superlu_dist_options_t & options,
+                      integer const & logLevel,
+                      real64 & timeFact,
+                      real64 & timeSolve )
+{
+  Stopwatch watch;
+
+  int_t const m = SLUDMat.nrow;
+  int_t const n = SLUDMat.ncol;
+
+  // Initialize ScalePermstruct.
+  dScalePermstruct_t ScalePermstruct;
+  dScalePermstructInit( m, n, &ScalePermstruct );
+
+  // Initialize LUstruct.
+  dLUstruct_t LUstruct;
+  dLUstructInit( n, &LUstruct );
+
+  // Initialize the statistics variables.
+  SuperLUStat_t stat;
+  PStatInit( &stat );
+
+  // Create process grid.
+  int const num_procs = MpiWrapper::Comm_size( comm );
+  int pcols = 1;
+  int prows = 1;
+  while( prows*pcols <= num_procs )
+  {
+    ++prows;
+  }
+  --prows;
+  pcols = num_procs/prows;
+  while( prows*pcols != num_procs )
+  {
+    prows -= 1;
+    pcols = num_procs/prows;
+  }
+  gridinfo_t grid;
+  superlu_gridinit( comm, prows, pcols, &grid );
+
+  // Call the linear equation solver.
+  int nrhs = 0;
```

Comment:
Is `0` the correct value for `nrhs`?

Reply:
Yes, it is.
I want to have two phases: factorization + solution. During the factorization, there is no need for rhs.